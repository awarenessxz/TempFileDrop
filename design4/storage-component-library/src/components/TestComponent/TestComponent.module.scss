.testComponent {
    background-color: white;
    border: 1px solid black;
    padding: 16px;
    text-align: center;

    .heading {
        font-size: 64px;
    }

    &.testComponentSecondary {
        background-color: black;
        color: white;
    }
}
