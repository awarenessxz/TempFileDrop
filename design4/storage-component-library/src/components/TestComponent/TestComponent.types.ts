export interface TestComponentProps {
    /** Theme of Test component */
    theme?: 'primary' | 'secondary';
    /** API to call */
    apiUrl?: string;
}
