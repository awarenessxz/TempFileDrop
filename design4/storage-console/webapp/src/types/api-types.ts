import { Moment } from "moment";

export interface UserTokenResponse {
    name: string;
    username: string;
    roles: string[];
    token: string;
    buckets: string[];
    routingKeys: string[];
}

export interface UserToken extends UserTokenResponse{
    isAdmin: boolean;
}

export interface FileSystemNode {
    isFile: boolean;
    label: string;
    storageSize: number;
    storageFullPath: string;
    storageBucket: string;
    storageDownloadLeft: number;
    storageExpiryDatetime?: Moment | null;
    children: FileSystemNode[];
}

export interface EventData {
    bucket: string;
    objectName: string;
    eventType: string;
    publishedDateTime: Moment;
}

export enum SchedulerJobStatus {
    COMPLETED = "COMPLETED",
    COMPLETED_WITH_ERROR = "COMPLETED_WITH_ERROR",
    PAUSED = "PAUSED",
    PENDING = "PENDING",
    RESCHEDULED = "RESCHEDULED",
    RESUMED = "RESUMED",
    RUNNING = "RUNNING",
    SCHEDULED = "SCHEDULED"
}

export enum SchedulerJobType {
    MONITOR_OBJECT = "MONITOR_OBJECT",
    MONITOR_OBJECT_CRONJOB = "MONITOR_OBJECT_CRONJOB"
}

export interface SchedulerJob {
    jobType: SchedulerJobType;
    jobName: string;
    description: string;
    cronExpression: string;
    startTime: Moment;
}

export interface WatchListJob {
    jobType: string;
    jobName: string;
    description: string;
    cronExpression: string;
    jobStatus: SchedulerJobStatus;
    isObjectValid: boolean;
}
