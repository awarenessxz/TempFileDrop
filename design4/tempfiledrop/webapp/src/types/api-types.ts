export interface UserTokenResponse {
    name: string;
    username: string;
    roles: string[];
    token: string;
    buckets: string[];
    routingKeys: string[];
}

export interface UserToken extends UserTokenResponse{
    isAdmin: boolean;
}