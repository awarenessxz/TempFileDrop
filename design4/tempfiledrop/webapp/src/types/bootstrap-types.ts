export interface AlertMessage {
    message: string;
    isError: boolean;
}