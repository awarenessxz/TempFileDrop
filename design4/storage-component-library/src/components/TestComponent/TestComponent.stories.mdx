import { Meta, Story, ArgsTable } from '@storybook/addon-docs/blocks';
import TestComponent from "./TestComponent";

<Meta title='Docs/TestComponent' component={TestComponent} />

# Test Component

If you have more information which you would like to add about your component, you can document it here

## Below are some possible documentation that you can do

### 1 - Display your story

*Note: The id is obtained from the url when you open storybook*

<Story id="components-testcomponent--primary" />

### 2 - Provide a Quick Start using simple markdown

```tsx
import TestComponent from "./TestComponent";

const test = () => {
    return <TestComponent theme='primary' />
};
```

### 3 - Display API / Interface

<ArgsTable of={TestComponent} />
