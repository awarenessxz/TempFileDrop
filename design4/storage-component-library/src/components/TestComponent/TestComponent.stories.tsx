import * as React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import TestComponent from './TestComponent';

export default {
    title: 'Components/TestComponent',
    component: TestComponent,
} as Meta;

/** *************************************************
 * Stories
 ************************************************** */

export const Primary = (): JSX.Element => <TestComponent theme="primary" apiUrl="/api/test/data" />;

export const Secondary = (): JSX.Element => <TestComponent theme="secondary" apiUrl="/api/test/data" />;
