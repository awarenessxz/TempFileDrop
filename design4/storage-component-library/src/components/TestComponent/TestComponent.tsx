import React, { MouseEvent, useState } from 'react';
import { Button } from "@material-ui/core";
import { TestComponentProps } from './TestComponent.types';
import styles from './TestComponent.module.scss';

/**
 * Test Component
 */
const TestComponent = ({
    theme = "primary",
    apiUrl = "/api/test/data"
}: TestComponentProps): JSX.Element => {
    const [apiMsg, setApiMsg] = useState("");
    const mThemeStyle = theme === 'primary' ? '' : styles.testComponentSecondary;

    const onButtonClick = (e: MouseEvent<HTMLButtonElement>) => {
        fetch(apiUrl)
            .then((res) => res.text())
            .then((data) => setApiMsg(data))
            .catch((err) => setApiMsg(err));
    };

    return (
        <div data-testid="test-component" className={`${styles.testComponent} ${mThemeStyle}`}>
            <h1 className={`${styles.heading}`}>I&apos;m the test component</h1>
            <h2>Made with love by Harvey</h2>
            <Button data-testid="test-button" variant="contained" color="primary" onClick={onButtonClick}>Call API</Button>
            {apiMsg && <p>{apiMsg}</p>}
        </div>
    );
};

export default TestComponent;
