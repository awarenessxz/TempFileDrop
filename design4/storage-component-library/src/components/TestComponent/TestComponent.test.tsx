import * as React from 'react';
import 'whatwg-fetch';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { render, RenderResult, screen } from '@testing-library/react';

// Import Component
import TestComponent from './TestComponent';
import { TestComponentProps } from './TestComponent.types';

/*********************************************************************
 * Configuration / Helper Functions
 ********************************************************************/

// set up mock server for rest api end point
const server = setupServer(
    rest.get('/mock/test/data', (req, res, ctx) => {
        return res(ctx.text('Hello, Mock Data!'));
    }),
);
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

// function to render Component before each test
type PartialTestComponentProps = Partial<TestComponentProps>;
const renderComponent = ({ ...props }: PartialTestComponentProps = {}): RenderResult => {
    const defaultProps: TestComponentProps = {
        theme: 'primary',
        apiUrl: '/mock/test/data'
    };
    const merged = { ...defaultProps, ...props };
    return render(<TestComponent {...merged} />);
};

/*********************************************************************
 * Test Cases
 ********************************************************************/

// 1. Testing if component renders properly
describe('Testing if component renders properly', () => {
    // Snapshot Testing
    it('Snapshot Testing', () => {
        const { asFragment } = renderComponent();
        expect(asFragment()).toMatchSnapshot();
    });

    // Testing Dom Elements: check if renders with correct value (Note: the usage of data-testid in component)
    describe('Testing if component renders with correct state/value', () => {
        it('should have primary className with default props', () => {
            const { getByTestId } = renderComponent();
            const testComponent = getByTestId('test-component');
            expect(testComponent).not.toHaveClass('testComponentSecondary');
        });

        it('should have secondary className with theme set as secondary', () => {
            const { getByTestId } = renderComponent({ theme: 'secondary' });
            const testComponent = getByTestId('test-component');
            expect(testComponent).toHaveClass('testComponentSecondary');
        });
    });
});

// 2. Testing Event Listener
describe('Testing OnClick Events', () => {
    it('Testing API Button', async () => {
        renderComponent();
        screen.getByTestId('test-button').click()
        const message = await screen.findByText('Hello, Mock Data!');
        expect(message).toBeInTheDocument();
    });
});
