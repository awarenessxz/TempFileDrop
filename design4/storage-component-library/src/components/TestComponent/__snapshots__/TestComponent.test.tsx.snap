// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if component renders properly Snapshot Testing 1`] = `
<DocumentFragment>
  <div
    class="testComponent "
    data-testid="test-component"
  >
    <h1
      class="heading"
    >
      I'm the test component
    </h1>
    <h2>
      Made with love by Harvey
    </h2>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-testid="test-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Call API
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
